### ISSUE TITLE
CGResult that saves info

### ISSUE THREAD

**Issue by mohamed82008** [JuliaMath/IterativeSolvers.jl#236]
It would be nice to have a `CGResult` struct that saves things like the last residual, number of iterations needed to converge, and options. If this is approved in principle, I can make a PR. This will be a breaking change however. A similar thing may also need to be done for the other algorithms but I am mostly interested in CG now.

**Comment by haampie** [comment]
Yes, since forever `cg(..., log = true)` has been type unstable as well. See https://github.com/JuliaMath/IterativeSolvers.jl/issues/185

**Comment by mohamed82008** [comment]
Cool, will make a PR then.
